stan/math/rev/mat/fun/Eigen_NumTraits.hpp

```diff
@@ -79,33 +79,32 @@ struct NumTraits<stan::math::var> : GenericNumTraits<stan::math::var> {
   static int digits10() { return std::numeric_limits<double>::digits10; }
 };
 
-namespace internal {
 /**
- * Partial specialization of Eigen's remove_all struct to stop
- * Eigen removing pointer from vari* variables
+ * Scalar product traits specialization for Eigen for reverse-mode
+ * autodiff variables.
  */
-template <>
-struct remove_all<stan::math::vari*> {
-  typedef stan::math::vari* type;
+template <typename BinaryOp>
```

Comment:
Note: Github confused me here with the way the changes look here as it looks like you removed the remove_all partial specialization but its then then added below.